irohad/main/application.cpp

```diff
@@ -182,47 +205,76 @@ void Irohad::initPeerCommunicationService() {
   log_->info("[Init] => pcs");
 }
 
-void Irohad::initTransactionCommandService() {
+void Application::initTransactionCommandService() {
+  PRECONDITION_TRUE(pcs);
+  PRECONDITION_TRUE(stateless_validator);
+  PRECONDITION_TRUE(pb_tx_factory);
+
   auto tx_processor =
       std::make_shared<TransactionProcessorImpl>(pcs, stateless_validator);
 
+  POSTCONDITION_TRUE(tx_processor);
+
   command_service =
       std::make_unique<::torii::CommandService>(pb_tx_factory, tx_processor);
+  POSTCONDITION_TRUE(command_service);
 
   log_->info("[Init] => command service");
 }
 
-void Irohad::initQueryService() {
+void Application::initQueryService() {
+  PRECONDITION_TRUE(storage);
+  PRECONDITION_TRUE(stateless_validator);
+  PRECONDITION_TRUE(pb_query_factory);
+  PRECONDITION_TRUE(pb_query_response_factory);
+
   auto query_proccessing_factory = std::make_unique<QueryProcessingFactory>(
       storage->getWsvQuery(), storage->getBlockQuery());
 
+  POSTCONDITION_TRUE(query_proccessing_factory);
+
   auto query_processor = std::make_shared<QueryProcessorImpl>(
       std::move(query_proccessing_factory), stateless_validator);
 
+  POSTCONDITION_TRUE(query_processor);
+
   query_service = std::make_unique<::torii::QueryService>(
       pb_query_factory, pb_query_response_factory, query_processor);
 
+  POSTCONDITION_TRUE(query_service);
+
   log_->info("[Init] => query service");
 }
 
-void Irohad::run() {
-  torii_server =
-      std::make_unique<ServerRunner>("0.0.0.0:" + std::to_string(torii_port_));
+void Application::run(const Torii &torii) {
+  torii_server = std::make_unique<ServerRunner>(torii.listenAddress());
+  POSTCONDITION_TRUE(torii_server);
 
   grpc::ServerBuilder builder;
-  int port = 0;
-  builder.AddListeningPort("0.0.0.0:" + std::to_string(internal_port_),
-                           grpc::InsecureServerCredentials(),
-                           &port);
+  int *port{};
```

Comment:
becaue `AddListeningPort` sets 3rd arg to either number, or nullptr. It should be pointer, otherwise there is nullptr -> int implicit cast.

Reply:
The [documentation](https://grpc.io/grpc/cpp/classgrpc_1_1_server_builder.html#a5bf2d8c40e6b4ca170b9ee927d4db326) states different behaviour:
>If not nullptr, gets populated with the port number bound to the grpc::Server for the corresponding endpoint after it is successfully bound, 0 otherwise. 

Therefore, it is required to pass a pointer to allocated int like it was done before.

Reply:
@lebdron 
![image](https://user-images.githubusercontent.com/1867551/32160698-e7d3d83e-bd63-11e7-9e52-0bef2ad667b0.png)

Documentation confused me.